library/src/layout/par.rs

```diff
@@ -998,15 +1003,65 @@ fn linebreak_optimized<'a>(vt: &Vt, p: &'a Preparation<'a>, width: Abs) -> Vec<L
     lines
 }
 
+/// Generated by the following command:
+///
+/// ```sh
+/// icu4x-datagen  --locales full --keys-for-bin target/debug/typst \
+///                --format blob --out assets/data/icudata.postcard --overwrite
+/// ```
+///
+/// Install icu4x-datagen with `cargo install icu4x-datagen`.
+static ICU_DATA: &[u8] = include_bytes!("../../../assets/data/icudata.postcard");
+
+/// Gnerated by the following command:
+///
+/// ```sh
+/// icu4x-datagen --locales zh ja --keys segmenter/line@1 --format blob \
+///               --out assets/data/cj_linebreak_data.postcard --overwrite
+/// ```
+///
+/// The used icu4x-datagen should be patched by
+/// https://github.com/peng1999/icu4x/commit/b9beb6cbf633d61fc3d7983e5baf7f4449fbfae5
```

Comment:
what does this patch do? do you intend to upstream it to icu4x?

Reply:
This generates data suitable for the Chinese/Japanese linebreak algorithm, using tailored rules from the ICU library. See unicode-org/icu#223.

This patch is a temporary solution and will not be included upstream. The developers have plans to support locale inside icu_segmenter in version 2.0. See unicode-org/icu4x#3284